habitat_sim/utils/data/dataextractor.py

```diff
@@ -0,0 +1,283 @@
+    iteratively generate image data.
+
+    :property scene_filepath: The location of the .glb file given to the simulator
+    :property labels: class labels of things to tather images of
+    :property img_size: output image dimensions (Height, Width)
+    :property cfg: configuration for simulator of type SimulatorConfiguration
+    :property sim: Simulator object
+    :property res: Resolution of topdown map. 0.1 means each pixel in the topdown map
+        represents 0.1 x 0.1 meters in the coordinate system of the pathfinder
+    :property tdv: TopdownView object
+    :property topdown_view: The actual 2D array representing the topdown view
+    :property pose_extractor: PoseExtractor object
+    :property poses: list of camera poses gathered from pose_extractor
+    :property label_map: maps lable numbers on the topdown map to their name
+    :property out_name_to_sensor_name: maps name of output to the sensor same corresponding to that output
+    :property output: list of output names that the user wants e.g. ['rgb', 'depth']
+    """
+
+    def __init__(
+        self,
+        scene_filepath,
+        labels=[0.0],
+        img_size=(512, 512),
+        output=["rgb"],
+        sim=None,
+    ):
+        self.scene_filepath = scene_filepath
+        self.labels = set(labels)
+        self.img_size = img_size
+        self.cfg = self._config_sim(self.scene_filepath)
+
+        if sim is None:
+            sim = habitat_sim.Simulator(self.cfg)
+
+        self.sim = sim
+        self.res = 0.1
+        self.tdv = TopdownView(self.sim, self.res)
+        self.topdown_view = self.tdv.topdown_view
+
+        self.pose_extractor = PoseExtractor(
+            self.topdown_view, self.sim.pathfinder, self.res
+        )
+        self.poses = self.pose_extractor.extract_poses(
+            labels=self.labels
+        )  # list of poses
+        self.label_map = {0.0: "unnavigable", 1.0: "navigable"}
+
+        # Configure the output each data sample
+        self.out_name_to_sensor_name = {
+            "rgb": "color_sensor",
+            "depth": "depth_sensor",
+            "semantic": "semamtic_sensor",
+        }
+        self.output = output
+
+    def __len__(self):
+        return len(self.poses)
+
+    def __getitem__(self, idx):
+        if isinstance(idx, slice):
+            start, stop, step = idx.start, idx.stop, idx.step
+            if start is None:
+                start = 0
+            if stop is None:
+                stop = len(self.poses)
+            if step is None:
+                step = 1
+
+            return [
+                self.__getitem__(i)
+                for i in range(start, stop, step)
+                if i < len(self.poses)
+            ]
+
+        pos, rot, label = self.poses[idx]
+        new_state = AgentState()
+        new_state.position = pos
+        new_state.rotation = rot
+        self.sim.agents[0].set_state(new_state)
+        obs = self.sim.get_sensor_observations()
+        sample = {
```

Comment:
Color is output as RGB-A, so you'll want to either chop-off the alpha channel or change the name.